wrappers/python/simtk/openmm/app/forcefield.py

```diff
@@ -1674,6 +1674,171 @@ def postprocessSystem(self, sys, data, args):
+            generator = existing[0]
+
+        for LJ in element.findall('Atom'):
+            generator.registerLennardJones(LJ.attrib)
+        for Nbfix in element.findall('NBFixPair'):
+            generator.registerNBFIX(Nbfix.attrib)
+
+    def createForce(self, sys, data, nonbondedMethod, nonbondedCutoff, args):
+
+        #Ewald and PME will be interpreted as CutoffPeriodic for the CustomNonbondedForce
+
+        # We need a CustomNonbondedForce to implement the NBFIX functionality.
+        # First derive the lookup tables
+        lj_indx_list = [0 for atom in data.atoms]
+        num_lj_types= 0
+        lj_type_list = []
+        type_map = {}
+        for i, atom in enumerate(data.atoms):
+            atype = data.atomType[atom]
+            values = self.lj_types.paramsForType[atype]
+            if lj_indx_list[i]: continue # ljtype has been assigned an index
+            if values in type_map.values() and not atype in [type for key in self.nbfix_types.keys() for type in key]:
+                # only non NBFIX types can be compressed
+                lj_indx_list[i] = type_map.values().index(values) + 1
+            else:
+                type_map[num_lj_types + 1] = values
+                lj_indx_list[i] = num_lj_types + 1
+                num_lj_types +=1
+                lj_type_list.append(atype)
+
+        # Now everything is assigned. Create the A- and B-coefficient arrays
+        acoef = [0 for i in range(num_lj_types*num_lj_types)]
+        bcoef = acoef[:]
+        for m in range(num_lj_types):
+            for n in range(num_lj_types):
+                pair = (lj_type_list[m], lj_type_list[n])
+                if len(self.nbfix_types) > 0:
+                    if pair in self.nbfix_types:
+                        wdij = self.nbfix_types[pair][0]
+                        rij = self.nbfix_types[pair][1]
+                        rij6 = rij**6
+                        acoef[m+num_lj_types*n] = math.sqrt(wdij) * rij6
+                        bcoef[m+num_lj_types*n] = 2 * wdij * rij6
+                        continue
+                    else:
+                        rij = (type_map[m+1]['sigma'] + type_map[n+1]['sigma']) * 2**(1.0/6) / 2
+                        rij6 = rij**6
+                        wdij = math.sqrt(type_map[m+1]['epsilon'] * type_map[n+1]['epsilon'])
+                        acoef[m+num_lj_types*n] = math.sqrt(wdij) * rij6
+                        bcoef[m+num_lj_types*n] = 2 * wdij * rij6
+                else:
+                    rij = (type_map[m+1]['sigma'] + type_map[n+1]['sigma']) * 2**(1.0/6) / 2
+                    rij6 = rij**6
+                    wdij = math.sqrt(type_map[m+1]['epsilon'] * type_map[n+1]['epsilon'])
+                    acoef[m+num_lj_types*n] = math.sqrt(wdij) * rij6
+                    bcoef[m+num_lj_types*n] = 2 * wdij * rij6
+        self.force = mm.CustomNonbondedForce('(a/r6)^2-b/r6; r6=r2*r2*r2; r2=r^2; '
```

Comment:
You give me too much credit :).  This scheme is used in ParmEd because #628 (particularly [this comment](https://github.com/pandegroup/openmm/pull/628#issuecomment-56537401)) led me to believe that ^3 and ^2 specifically were optimized (since there's a `square` and `cube` function in Lepton not documented in the API docs, but apparently nothing higher).

So I thought that keeping things as ^2 was the "right" thing to do.  Also, in ParmEd, the `a` coefficient is sqrt-ed before added to the table.  Not sure what's done here...  If that's the wrong thing to do, I should adjust ParmEd lest it serve as a misleading example (as it seems it has here).